runtime/drivers/snowflake/snowflake.go

```diff
@@ -80,8 +82,18 @@ func (d driver) Open(instanceID string, config map[string]any, st *storage.Clien
 		return nil, err
 	}
 
-	// actual db connection is opened during query
+	if conf.DSN == "" {
+		return nil, fmt.Errorf("dsn not provided")
+	}
```

Comment:
Doesn't this break backwards compatibility with projects that configure the DSN in the sources (legacy way of configuring the DSN)? We definitely can't break backwards compatibility for any commonly used connectors.